Index: .idea/.gitignore
===================================================================
diff --git a/.idea/.gitignore b/.idea/.gitignore
deleted file mode 100644
--- a/.idea/.gitignore	(revision e362393dd8dd10969de0c6d5c1b02ed0964a8935)
+++ /dev/null	(revision e362393dd8dd10969de0c6d5c1b02ed0964a8935)
@@ -1,3 +0,0 @@
-# Default ignored files
-/shelf/
-/workspace.xml
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/modules.xml b/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1651456704035)
+++ b/.idea/modules.xml	(date 1651456704035)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/PythonProject.iml" filepath="$PROJECT_DIR$/PythonProject.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ProjectRootManager\" version=\"2\" project-jdk-name=\"Python 3.9\" project-jdk-type=\"Python SDK\">\r\n    <output url=\"file://$PROJECT_DIR$/out\" />\r\n  </component>\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	(revision e362393dd8dd10969de0c6d5c1b02ed0964a8935)
+++ b/.idea/misc.xml	(date 1651456703973)
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.9" project-jdk-type="Python SDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="Python 3.9" project-jdk-type="Python SDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>
\ No newline at end of file
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"VcsDirectoryMappings\">\r\n    <mapping directory=\"$PROJECT_DIR$\" vcs=\"Git\" />\r\n  </component>\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
--- a/.idea/vcs.xml	(revision e362393dd8dd10969de0c6d5c1b02ed0964a8935)
+++ b/.idea/vcs.xml	(date 1651456704098)
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    <mapping directory="" vcs="Git" />
   </component>
 </project>
\ No newline at end of file
Index: differential equation.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/differential equation.py b/differential equation.py
new file mode 100644
--- /dev/null	(date 1675626089915)
+++ b/differential equation.py	(date 1675626089915)
@@ -0,0 +1,49 @@
+import numpy as np
+from scipy import integrate
+import matplotlib.pyplot as plt
+import sympy
+
+
+def plot_direction_field(x, y_x, f_xy, x_lim=(-5, 5), y_lim=(-5, 5), ax=None):
+    f_np = sympy.lambdify((x, y_x), f_xy, 'numpy')
+    x_vec = np.linspace(x_lim[0], x_lim[1], 20)
+    y_vec = np.linspace(y_lim[0], y_lim[1], 20)
+
+    if ax is None:
+        _, ax = plt.subplots(figsize=(4, 4))
+
+    dx = x_vec[1] - x_vec[0]
+    dy = y_vec[1] - y_vec[0]
+
+    for m, xx in enumerate(x_vec):
+        for n, yy in enumerate(y_vec):
+            Dy = f_np(xx, yy) * dx
+            Dx = 0.8 * dx ** 2 / np.sqrt(dx ** 2 + Dy ** 2)
+            Dy = 0.8 * Dy * dy / np.sqrt(dx ** 2 + Dy ** 2)
+            ax.plot([xx - Dx / 2, xx + Dx / 2], [yy - Dy / 2, yy + Dy / 2], 'b', lw=0.5)
+
+    ax.axis('tight')
+    ax.set_title(r"$%s$" % (sympy.latex(sympy.Eq(y_x.diff(x), f_xy))), fontsize=18)
+
+    return ax
+
+
+x = sympy.symbols('x')
+y = sympy.Function('y')
+f = x - y(x) ** 2
+
+f_np = sympy.lambdify((y(x), x), f)
+# put variables (y(x), x) into lambda function f.
+y0 = 1
+xp = np.linspace(0, 5, 100)
+yp = integrate.odeint(f_np, y0, xp)
+# solve f_np with initial conditons y0, and x ranges as xp.
+xn = np.linspace(0, -5, 100)
+yn = integrate.odeint(f_np, y0, xn)
+
+fig, ax = plt.subplots(1, 1, figsize=(4, 4))
+plot_direction_field(x, y(x), f, ax=ax)
+# plot direction field of function f
+ax.plot(xn, yn, 'b', lw=2)
+ax.plot(xp, yp, 'r', lw=2)
+plt.show()
Index: PythonProject.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/PythonProject.iml b/PythonProject.iml
new file mode 100644
--- /dev/null	(date 1651456704067)
+++ b/PythonProject.iml	(date 1651456704067)
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
